sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md

```diff
@@ -0,0 +1,200 @@
+
+```C# Snippet:GeographicalAreaCodes
+var locationOptionsResponse = client.GetPhonePlanLocationOptions(countryCode, geographicPhonePlanGroupId, geographicPhonePlanId);
+var state = locationOptionsResponse.Value.LocationOptions.Options.First();
+
+var locationOptionsQueries = new List<LocationOptionsQuery>
+{
+    new LocationOptionsQuery
+    {
+        LabelId = "state",
+        OptionsValue = state.Value
+    },
+    new LocationOptionsQuery
+    {
+        LabelId = "city",
+        OptionsValue = state.LocationOptions.First().Options.First().Value
+    }
+};
+
+var areaCodes = client.GetAllAreaCodes(geographicPhonePlan.LocationType.ToString(), countryCode, geographicPhonePlan.PhonePlanId, locationOptionsQueries);
+
+foreach (var areaCode in areaCodes.Value.PrimaryAreaCodes)
+{
+    Console.WriteLine($"Primary area code: {areaCode}");
+}
+```
+
+Area codes for toll-free phone plans can be found in the plan.
+
+```C# Snippet:TollFreePlanAreaCodes
+var phonePlans = client.GetPhonePlans(countryCode, tollFreePhonePlanGroupId, locale);
+var tollFreePhonePlan = phonePlans.First();
+
+foreach (var areaCode in tollFreePhonePlan.AreaCodes)
+{
+    Console.WriteLine($"Area code: {areaCode}");
+}
+```
+
+## Reserve phone numbers
+
+Phone numbers need to be reserved before they can be purchased. Reservation is a long running operation that can be started by `CreateReservationOptions` function that returns an `PhoneNumberReservationOperation` object. `PhoneNumberReservationOperation` can be used to update status of the operation and to check for completeness.
+
+```C# Snippet:ReservePhoneNumbers
+var reservationName = "My reservation";
+var reservationDescription = "reservation description";
+var reservationOptions = new CreateReservationOptions(reservationName, reservationDescription, new[] { phonePlanId }, areaCode);
+reservationOptions.Quantity = 1;
+
+var reservationOperation = client.StartReservation(reservationOptions);
+
+while (!reservationOperation.HasCompleted)
+{
+    Thread.Sleep(2000);
+
+    reservationOperation.UpdateStatus();
+}
+```
+
+## Persist reserve phone numbers operation
+
+You can persist the operation Id of the phone number reservation so that you can come back and check the operation status later.
+
+```C# Snippet:PersistReservePhoneNumbersOperation
+var reservationId = reservationOperation.Id;
+
+// persist reservationOperationId and then continue with new operation
+
+var newPhoneNumberReservationOperation = new PhoneNumberReservationOperation(client, reservationId);
+
+while (!newPhoneNumberReservationOperation.HasCompleted)
+{
+    Thread.Sleep(2000);
+
+    newPhoneNumberReservationOperation.UpdateStatus();
+}
+```
+
+## Purchase phone numbers
+
+Phone numbers can be acquired through purchasing a reservation.
+
+```C# Snippet:StartPurchaseReservation
+var reservationPurchaseOperation = client.StartPurchaseReservation(reservationId);
```

Comment:
We can shorten this to `purchaseOperation`

Reply:
Done